In [1]:
import pandas as pd

In [3]:
# we can import data in a wide variety of formats. Like read_sql, read_json, read_excel, read_csv etc.
oo = pd.read_csv('data/olympics.csv')
oo.head()

List of medallists at the Games of the Olympiad per edition, sport, discipline, gender and event  \
0                                                NaN                                                 
1  DISCLAIMER: The IOC Research and Reference Ser...                                                 
2                                                NaN                                                 
3                                               City                                                 
4                                             Athens                                                 

  Unnamed: 1 Unnamed: 2  Unnamed: 3     Unnamed: 4 Unnamed: 5 Unnamed: 6  \
0        NaN        NaN         NaN            NaN        NaN        NaN   
1        NaN        NaN         NaN            NaN        NaN        NaN   
2        NaN        NaN         NaN            NaN        NaN        NaN   
3    Edition      Sport  Discipline        Athlete        NOC     Gender   
4       1896   Aquatics    Swimming  HAJOS, Alfred        HUN        Men   

       Unnamed: 7    Unnamed: 8 Unnamed: 9  
0             NaN           NaN        NaN  
1             NaN           NaN        NaN  
2             NaN           NaN        NaN  
3           Event  Event_gender      Medal  
4  100m freestyle             M       Gold

In [4]:
# first few rows dont seem to make sense. If we open the csv file in notepad++(with line numbers), 
# we will see that we need to skip first 4 rows. 
# You can also open it in Excel instead of Notepad++
oo = pd.read_csv('data/olympics.csv', skiprows=4)
oo.head()

City  Edition     Sport Discipline             Athlete  NOC Gender  \
0  Athens     1896  Aquatics   Swimming       HAJOS, Alfred  HUN    Men   
1  Athens     1896  Aquatics   Swimming    HERSCHMANN, Otto  AUT    Men   
2  Athens     1896  Aquatics   Swimming   DRIVAS, Dimitrios  GRE    Men   
3  Athens     1896  Aquatics   Swimming  MALOKINIS, Ioannis  GRE    Men   
4  Athens     1896  Aquatics   Swimming  CHASAPIS, Spiridon  GRE    Men   

                        Event Event_gender   Medal  
0              100m freestyle            M    Gold  
1              100m freestyle            M  Silver  
2  100m freestyle for sailors            M  Bronze  
3  100m freestyle for sailors            M    Gold  
4  100m freestyle for sailors            M  Silver

#### Shape
- It is good to validate the read data by using chape and then cross-checking with the number of rows in the source data file
- shape returns a tuple.. rows and columns in the read data
- if you want only row or column, you can extract via indexes 0 or 1

In [7]:
oo.shape

(29216, 10)

In [8]:
# only row
print(oo.shape[0])
# only cols
print(oo.shape[1])

29216
10


#### head, tail

In [18]:
print('head:\n',oo.head())
print("\n")
print("\n")
print('tail:\n',oo.tail())
print("\n")
print("\n")
print('head:\n',oo.head(3))
print("\n")
print("\n")
print('tail:\n',oo.tail(3))

head:
      City  Edition     Sport Discipline             Athlete  NOC Gender  \
0  Athens     1896  Aquatics   Swimming       HAJOS, Alfred  HUN    Men   
1  Athens     1896  Aquatics   Swimming    HERSCHMANN, Otto  AUT    Men   
2  Athens     1896  Aquatics   Swimming   DRIVAS, Dimitrios  GRE    Men   
3  Athens     1896  Aquatics   Swimming  MALOKINIS, Ioannis  GRE    Men   
4  Athens     1896  Aquatics   Swimming  CHASAPIS, Spiridon  GRE    Men   

                        Event Event_gender   Medal  
0              100m freestyle            M    Gold  
1              100m freestyle            M  Silver  
2  100m freestyle for sailors            M  Bronze  
3  100m freestyle for sailors            M    Gold  
4  100m freestyle for sailors            M  Silver  




tail:
           City  Edition      Sport       Discipline               Athlete  \
29211  Beijing     2008  Wrestling  Wrestling Gre-R        ENGLICH, Mirko   
29212  Beijing     2008  Wrestling  Wrestling Gre-R  MIZGAI

### info
- Info provides a summary of the data frame including the number of entries, the data type, and the number of non-null entries for each series in the data frame. 
- This is important because often when working with a real data set, there will be missing data. You want a view of this to determine how you will handle this missing data.

In [19]:
oo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29216 entries, 0 to 29215
Data columns (total 10 columns):
City            29216 non-null object
Edition         29216 non-null int64
Sport           29216 non-null object
Discipline      29216 non-null object
Athlete         29216 non-null object
NOC             29216 non-null object
Gender          29216 non-null object
Event           29216 non-null object
Event_gender    29216 non-null object
Medal           29216 non-null object
dtypes: int64(1), object(9)
memory usage: 2.2+ MB


- We can see that we do not have any missing data in this data set.
- There are 29,216 entries, and all of them are non-null.
- What is also useful about the info method is that it also gives you the data types for the different series of columns. This is useful when you have to do any comparisons or queries. So for example, if you are querying what is a string, then you will know that you have to use quotes, and if you are using a comparison for an integer or a float value, then you won't need to use those quotes.